rewriter/src/main/java/com/facebook/coresql/rewriter/Rewriter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.rewriter;
+
+import com.facebook.coresql.parser.AstNode;
+import com.facebook.coresql.parser.Unparser;
+
+import static com.facebook.coresql.parser.ParserHelper.parseStatement;
+import static java.util.Objects.requireNonNull;
+
+public abstract class Rewriter
+        extends Unparser
+{
+    private boolean isEnabled;
+
+    public Rewriter()
+    {
+        this.isEnabled = true;
+    }
+
+    abstract RewriteResult rewrite(String originalSql);
+
+    public static AstNode validateThenParseStatement(String sql)
+    {
+        AstNode sqlAsAst = parseStatement(requireNonNull(sql));
+        return requireNonNull(sqlAsAst);
+    }
+
+    public boolean isEnabled()
```

Comment:
This is not used. Are you supposed to use it in your order by rewriter? If you are not planning to use it in this PR, remove it.

Reply:
This is part of the API. Every rewriter will have a isEnabled that can be controlled by config.

Reply:
After off-line chat, I think we can remove this whole thing for now and add it in the integration PR.

Reply:
Removed this field and its accessors/mutators.

Done